src/node_interface.rs

```diff
@@ -40,7 +50,7 @@ pub trait Interface : Sync + Send {
                       from_authority: Authority,
                       from_address: NameType) -> Result<MessageAction, InterfaceError>;
 
-    /// if reply is data then we send back the response message (ie get_response )
+    /// data or address of the node potentially storing data is returned on success.
```

Comment:
the data is not returned on success; the return value is the action routing is asked to perform on the message to continue the message flow.